python/ray/rllib/agents/impala/vtrace.py

```diff
@@ -200,34 +128,29 @@ def multi_from_logits(behaviour_policy_logits,
       target_action_log_probs: A float32 tensor of shape [T, B] containing
         target policy action probabilities (log \pi(a_t)).
   """
-
-    for i in range(len(behaviour_policy_logits)):
-        behaviour_policy_logits[i] = tf.convert_to_tensor(
-            behaviour_policy_logits[i], dtype=tf.float32)
-        target_policy_logits[i] = tf.convert_to_tensor(
-            target_policy_logits[i], dtype=tf.float32)
-        actions[i] = tf.convert_to_tensor(actions[i], dtype=tf.int32)
-
-        # Make sure tensor ranks are as expected.
-        # The rest will be checked by from_action_log_probs.
-        behaviour_policy_logits[i].shape.assert_has_rank(3)
-        target_policy_logits[i].shape.assert_has_rank(3)
-        actions[i].shape.assert_has_rank(2)
+    behaviour_policy_logits = tf.convert_to_tensor(
+        behaviour_policy_logits, dtype=tf.float32)
+    target_policy_logits = tf.convert_to_tensor(
+        target_policy_logits, dtype=tf.float32)
+    actions = tf.convert_to_tensor(actions, dtype=tf.int32)
+
+    # Make sure tensor ranks are as expected.
+    # The rest will be checked by from_action_log_probs.
+    behaviour_policy_logits.shape.assert_has_rank(3)
+    target_policy_logits.shape.assert_has_rank(3)
+    actions.shape.assert_has_rank(2)
 
     with tf.name_scope(
             name,
             values=[
                 behaviour_policy_logits, target_policy_logits, actions,
                 discounts, rewards, values, bootstrap_value
             ]):
-        target_action_log_probs = multi_log_probs_from_logits_and_actions(
+        target_action_log_probs = log_probs_from_logits_and_actions(
             target_policy_logits, actions)
-        behaviour_action_log_probs = multi_log_probs_from_logits_and_actions(
+        behaviour_action_log_probs = log_probs_from_logits_and_actions(
             behaviour_policy_logits, actions)
-
-        log_rhos = get_log_rhos(target_action_log_probs,
-                                behaviour_action_log_probs)
-
+        log_rhos = target_action_log_probs - behaviour_action_log_probs
```

Comment:
One potential source of the bug is if this guy has a wrong shape.

Reply:
I checked the shapes here and it looked reasonable (at least they match).